data/gamemoderun.in

```diff
@@ -1,11 +1,44 @@
 #!/bin/bash
 # Helper script to launch games with gamemode
 
+# Load vsync mode via dbus
+vsync_mode="$(dbus-send --session --dest=com.feralinteractive.GameMode --type=method_call --print-reply /com/feralinteractive/GameMode com.feralinteractive.GameMode.GetVsyncMode | grep -oP '(?<=").*(?=")')"
+
+# Load hybrid GPU mode via dbus
+hybrid_gpu_mode="$(dbus-send --session --dest=com.feralinteractive.GameMode --type=method_call --print-reply /com/feralinteractive/GameMode com.feralinteractive.GameMode.GetHybridGPUMode | grep -oP '(?<=").*(?=")')"
+
+# Set vsync mode
+if [ "$vsync_mode" == "force_disable" ]; then
+  echo "gamemoderun: force disable vsync"
+  export vblank_mode=0
+elif [ "$vsync_mode" == "force_enable" ]; then
+  echo "gamemoderun: force enable vsync"
+  export vblank_mode=1
+fi
+
+# Set hybrid GPU mode
+if [ "$hybrid_gpu_mode" == "prime" ]; then
+  echo "gamemoderun: launch with prime"
+  exec_cmd="exec"
+  export DRI_PRIME=1
+elif [ "$hybrid_gpu_mode" == "bumblebee" ]; then
+  echo "gamemoderun: launch with bumblebee"
+  exec_cmd="optirun"
+elif [ "$hybrid_gpu_mode" == "primus" ]; then
+  echo "gamemoderun: launch with primus"
+  exec_cmd="primusrun"
+elif [ "$hybrid_gpu_mode" == "nvidia-xrun" ]; then
+  echo "gamemoderun: launch with nvidia-xrun"
+  exec_cmd="nvidia-xrun"
+else
+  exec_cmd="exec"
```

Comment:
`exec` is used in the script to ensure the whole process is replaced. For consistency it seems like we should still always call exec, but with the additional arguments as wrappers:
```
exec "$hybrid_gpu_launcher" "$@"
```

Reply:
I think thats a good idea. We can simply use a command in `main.c` to output all env vars like `vblank_mode` or `DRI_PRIME`, and then export all of them line by line and use another one to output additional launch commands. Sadly it dosn't look like it's possible to set env vars in the launch command via the script, at least the way I tried.